lib/runtime/errors/unknown_transaction_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package errors
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_UnknownTransaction_Errors(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name        string
+		encodedData []byte
+		expErr      bool
+		expErrMsg   string
+		expValidity *transaction.Validity
+	}{
+		{
+			name:        "lookup failed",
+			encodedData: []byte{1, 1, 0},
+			expErrMsg:   fmt.Errorf("%w: %s", ErrUnknownTxn, "lookup failed").Error(),
+			expErr:      true,
+		},
+	}
+
+	for _, c := range testCases {
+		c := c
+		t.Run(c.name, func(t *testing.T) {
+			t.Parallel()
+			validity, err := UnmarshalTransactionValidity(c.encodedData)
+			if !c.expErr {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.EqualError(t, err, c.expErrMsg)
+			}
+
+			require.Equal(t, c.expValidity, validity)
```

Comment:
```suggestion
			validity, err := UnmarshalTransactionValidity(c.encodedData)
			if !c.expErr {
				require.NoError(t, err)
				require.Equal(t, c.expValidity, validity)
			} else {
				require.Error(t, err)
				require.EqualError(t, err, c.expErrMsg)
				require.Nil(t, validity)
			}

			
```

Reply:
I preferred it before. The validity is check against `c.expValidity` which is `nil` when it should be.

Reply:
Could you elaborate on why this is preferred?

Reply:
I suggested ensuring `validity` was nil when an error occur but I see that the current way you already check that by setting `c.expValidity` as nil, so nevermind 😅